src/OrchardCore.Modules/OrchardCore.GitHub/Configuration/GithubOptionsConfiguration.cs

```diff
@@ -53,22 +59,26 @@ public void Configure(AuthenticationOptions options)
         public void Configure(string name, GitHubOptions options)
         {
             // Ignore OpenID Connect client handler instances that don't correspond to the instance managed by the OpenID module.
-            if (!string.Equals(name, GitHubDefaults.AuthenticationScheme))
+            if (!String.Equals(name, GitHubDefaults.AuthenticationScheme))
             {
                 return;
             }
 
             var loginSettings = GetGitHubAuthenticationSettingsAsync().GetAwaiter().GetResult();
+            if (loginSettings == null)
+            {
+                return;
+            }
 
-            options.ClientId = loginSettings?.ClientID ?? string.Empty;
+            options.ClientId = loginSettings.ClientID;
```

Comment:
Maybe at least keep a null check on ClientID?

Reply:
`ClientID` is checked in `ValidateSettings()` that is called by the above `GetGitHubAuthenticationSettingsAsync()` that returns null if the validation fails